test/powershell/engine/ValidateAttributes.Tests.ps1

```diff
@@ -22,7 +46,270 @@ Describe 'Validate Attributes Tests' -Tags 'CI' {
+                   FullyQualifiedErrorId = "MaxRangeSmallerThanMinRange"
+                   InnerErrorId = "" 
+                }
+               @{ 
+                   sb = { function foo { param([ValidateRange("one",10)] $bar) }; foo }
+                   FullyQualifiedErrorId = "MinRangeNotTheSameTypeOfMaxRange"
+                   InnerErrorId = "" 
+                }
+                @{ 
+                   sb = { function foo { param([ValidateRange(1,"two")] $bar) }; foo }
+                   FullyQualifiedErrorId = "MinRangeNotTheSameTypeOfMaxRange"
+                   InnerErrorId = "" 
+                }
+           )
+        }
+
+        It 'Exception: <FullyQualifiedErrorId>:<InnerErrorId>' -TestCases $testCases {
+            param($sb, $FullyQualifiedErrorId, $InnerErrorId)
+
+            $sb | ShouldBeErrorId $FullyQualifiedErrorId
+            if ($InnerErrorId) {
+                $error[0].exception.innerexception.errorrecord.FullyQualifiedErrorId | Should Be $InnerErrorId
+            }
+        }
+    }
+    Context "ValidateRange - Range"{
+        BeforeAll {
+           $testCases = @(
+               @{ 
+                   sb = { function foo { param([ValidateRange(1,10)] [int] $bar) }; foo -1 }
+                   FullyQualifiedErrorId = "ParameterArgumentValidationError,foo"
+                   InnerErrorId = "ValidateRangeTooSmall" 
+                }
+               @{ 
+                   sb = { function foo { param([ValidateRange(1,10)] [int] $bar) }; foo 11 }
+                   FullyQualifiedErrorId = "ParameterArgumentValidationError,foo"
+                   InnerErrorId = "ValidateRangeTooBig" 
+                }
+               @{ 
+                   sb = { function foo { param([ValidateRange(1,10)] $bar) }; foo "one" }
+                   FullyQualifiedErrorId = "ParameterArgumentValidationError,foo"
+                   InnerErrorId = "ValidationRangeElementType" 
+                }
+           )
+
+            $validTestCases = @(
+               @{ 
+                   sb = { function foo { param([ValidateRange(1,10)] [int] $bar) }; foo 5 }
+                }
+           )
+        }
+
+        It 'Exception: <FullyQualifiedErrorId>:<InnerErrorId>' -TestCases $testCases {
+            param($sb, $FullyQualifiedErrorId, $InnerErrorId)
+
+            $sb | ShouldBeErrorId $FullyQualifiedErrorId
+            if ($InnerErrorId) {
+                $error[0].exception.innerexception.errorrecord.FullyQualifiedErrorId | Should Be $InnerErrorId
+            }
+        }
+
+        It 'No Exception: value within range' -TestCases $validTestCases {
+            param($sb)
+                $sb | Should Not Throw
+        }
+    }
+
+    Context "ValidateRange - Positive" {
+        BeforeAll {
+           $testCases = @(
+               @{ 
+                   sb = { function foo { param([ValidateRange('Positive')] [int] $bar) }; foo -1 }
```

Comment:
I tried this but it failed since ValidateRangeKind is not exposed as a type accelerator. Should i use the full namespace or add [ValidateRangeKind]?